docs/docs/subscription.fsx

```diff
@@ -160,12 +261,195 @@ module App =
+
+(**
+Notice `Sub.map` takes an id prefix as its first parameter. This helps keep subscriptions distinct from each other.
+
+Before Sub.map, Second and Hour have the same ID: `["timer"]`. After Sub.map, their IDs are: `["hour"; "timer"]` and `["second"; "timer"]`.
+*)
+
+(**
+### Subscription reusability with Effect
+
+> 📌 **Effect**
+> The `Effect` type was known as `Sub` in v3. With the change to subscriptions, Sub would have been an overloaded and confusing term.
+> Effect works exactly the same as before. Only the names have been changed to protect the innocent.
+
+In the last section, we increased reusability of the timer by taking the interval as a parameter.
+We can use Effect to factor out the hard-coded `DateTime.Now` behavior for even more reuse.
+
+First let's turn `DateTime.Now` into an Effect.
+*)
+
+module EffectTimer =
+
+    module Time =
+
+        let now (tag: DateTime -> 'msg) : Effect<'msg> =
+            let effectFn dispatch =
+                dispatch (tag DateTime.Now)
+            effectFn
+
+(**
+> Strange as it sounds, DateTime.Now _is_ a side effect. This property reads from a clock and may return a different value each time. Code that uses it becomes nondeterministic.
+
+Now let's change timer to run an Effect when the interval fires.
+*)
+
+    module Sub =
+
+        let timer intervalMs (effect: Effect<'msg>) =
+            let start dispatch =
+                let intervalId = 
+                    JS.setInterval (fun _ -> effect dispatch) intervalMs
+                { new IDisposable with
+                    member _.Dispose() = JS.clearInterval intervalId }
+            start
+
+(**
+And finally, here is the updated `subscribe` function.
+*)
+
+    type Model =
+        {
+            current : DateTime
+            intervalMs : int
+        }
+
+    type Msg =
+        | Tick of DateTime
+
+    let subscribe model =
+        [ ["timer"], Sub.timer model.intervalMs (Time.now Tick) ]
+
+(**
+The timer subscription can now run any kind of effect. Calling an API, playing a sound, whatever.
+
+`Time.now` is also usable for Cmds.
+*)
+
+    let init () =
+        {
+            current = DateTime.MinValue
+            intervalMs = 1000
+        }
+        , Cmd.ofEffect (Time.now Tick)
+
+(**
+### IDs and dependencies
+
+I mentioned earlier that ID is a list so that you can add dependencies to it. What does this mean?
+
+In the last example, the timer's interval came from the model:
+
+`Sub.timer model.intervalMs (Time.now Tick)`
+
+But nothing happens to the subscription if `model.intervalMs` changes. Let's fix that.
+*)
+
+(*** hide ***)
+module IdDeps =
+
+    open EffectTimer // reuse last example's code
+
+(*** show ***)
+    let subscribe model =
+        [ ["timer"; string model.intervalMs], Sub.timer model.intervalMs (Time.now Tick) ]
+
+(**
+Now that intervalMs is part of the ID, an interesting thing happens. When the interval changes, the timer will stop then restart with the new interval.
```

Comment:
I would remove "an interesting thing happens"

Reply:
It is only there to avoid multiple conditional phrases in a row. "Now that ..., when the ..." It's like code with nested if statements. Takes extra effort to understand. Terminating the first conditional phrase helps. Changing to a statement of fact: "`intervalMs` is part of the ID." is stating an obvious fact without making a direct connection to the next sentence. I left the sentence in as more of an introduction and segway. Then the concepts are explained in more detail in the next section. Writing can be quite a logic puzzle sometimes. Basically writing code for human brains. I'll try to find a different way to phrase it.